i18n-helpers/src/bin/markdown-xgettext.rs

```diff
@@ -0,0 +1,229 @@
+use anyhow::{bail, Context};
+use clap::{Arg, Command};
+use mdbook_i18n_helpers::{add_message, extract_messages};
+use polib::catalog::Catalog;
+use polib::metadata::CatalogMetadata;
+use polib::po_file;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+use std::{fs, io};
+
+fn create_catalog(
+    content: &str,
+    source_path: &str,
+    language: &str,
+    title: Option<&str>,
+) -> anyhow::Result<Catalog> {
+    let mut metadata = CatalogMetadata::new();
+
+    if let Some(title) = title {
+        metadata.project_id_version = String::from(title);
+    }
+
+    metadata.language = String::from(language);
+    let now = chrono::Local::now();
+    metadata.pot_creation_date = now.to_rfc3339_opts(chrono::SecondsFormat::Secs, true);
+    metadata.mime_version = String::from("1.0");
+    metadata.content_type = String::from("text/plain; charset=UTF-8");
+    metadata.content_transfer_encoding = String::from("8bit");
+    let mut catalog = Catalog::new(metadata);
+
+    for (lineno, msgid) in extract_messages(content) {
+        let source = format!("{}:{}", source_path, lineno);
+        add_message(&mut catalog, &msgid, &source);
+    }
+
+    Ok(catalog)
+}
+
+fn ingest_input(input_dir: Option<&str>) -> anyhow::Result<(&str, String)> {
+    if let Some(file_path) = input_dir {
+        let content = fs::read_to_string(file_path)
+            .context(format!("Failed to read from file {}", file_path))?;
+        Ok((file_path, content))
+    } else {
+        let mut buffer = String::new();
+        io::stdin()
+            .read_to_string(&mut buffer)
+            .context("Failed to read from stdin")?;
+        Ok(("-", buffer))
+    }
+}
+
+fn build_path(output_dir: Option<&str>, pot_title: &str) -> anyhow::Result<PathBuf> {
```

Comment:
I would like this function to be simpler:

- if `output` is `"-"`, then write to stdout
- otherwise write to the path given as output. People can create the necessary directories themselves.